tonic/datasets/s_mnist.py

```diff
@@ -0,0 +1,141 @@
+import os
+import numpy as np
+from struct import unpack
+from .dataset import Dataset
+from .download_utils import (
+    download_and_extract_archive,
+    extract_archive,
+)
+
+class SMNIST(Dataset):
+    """Spiking sequential MNIST"""
```

Comment:
could you please add a description of what SMNIST is? I think it would be helpful to know for newcomers how it's different from NMNIST for example. Also please add parameter descriptions, like here: https://github.com/neuromorphs/tonic/blob/d6ecb10a6d3a67392977d783760beca8abeaf29f/tonic/datasets/nmnist.py#L14-L24